src/NUnitEngine/nunit-agent/Program.cs

```diff
@@ -125,8 +125,8 @@ public static int Main(string[] args)
             log.Info("Initializing Services");
             engine.Initialize();
 
-            // Owns the channel used for communications with the agency and with clients
-            var testAgencyServer = engine.Services.GetService<TestAgency>();
+            // Make sure we can communicate with the agency
+            ServerUtilities.GetTcpChannel("", 0, 2);
```

Comment:
That's in a test, which includes TestAgency in a temporary context. Really, it should be using mocks or be moved to an integration test in my view.

The problem with the original code is that it uses a service (server) that allows the agent to create new agents, which is not supposed to exist in the agent. See my general comment.